tools/stats-definition-exporter/README.md

```diff
@@ -0,0 +1,83 @@
+# Sync Gateway Prometheus Stat Definition Exporter
+This tool exports all the Prometheus Sync Gateway stats definitions to a JSON file. The JSON is prettified by using tab indents and new lines.
+
+## Output
+### To File
+By default, this tool will output to a file. named `metrics_metadata.json`. This is outputted in the current working directory.
+
+Use the `-output` flag to change the output location and name.
+
+### To Stdout
+To output to stdout only and not to a fail, use the `-no-file` flag. The `-output` flag will be ignored when this is `true`.
+
+## Logging
+The tool may output to stderr when unexpected problems occur. For example, this maybe due to a stat not being initialized or a stat being nil.
+
+## JSON format
+The JSON is an array of objects in the current format:
+```json
+[
+  {
+    "name": "string",
+    "unit": "string",
+    "labels": ["strings"],
+    "help": "string",
+    "added": "string",
+    "stability": "string",
+    "type": "string"
+  }
+]
+```
+
+- `name` is the fully qualified name of the stat.
+- `unit` is what unit the stat uses such as bytes or nanoseconds. If the stat has no units, then this will be omitted.
+- `labels` is a list of label keys that Prometheus uses to uniquely distinguish between the same stat being declared multiple times. For example, `databases`, `collections` etc. This is omitted if the stat has no labels.
+- `help` contains a description of what the stat does.
+- `added` is the Sync Gateway version the stat got added.
+- `stability` is what the current stability of the stat is such as `committed`.
```

Comment:
Is this an enum? What are the other options?

Reply:
Yes, talked about it, he's adding them